frontends/ast/simplify.cc

```diff
@@ -2734,6 +2731,20 @@ bool AstNode::simplify(bool const_fold, bool at_zero, bool in_lvalue, int stage,
 			delete right_at_zero_ast;
 		}
 
+		int stride = 1;
+		AST::AstNode *member_node = get_struct_member(children[0]);
+		if (member_node) {
+			// Clamp chunk to range of member within struct/union.
+			log_assert(!source_offset && !children[0]->id2ast->range_swapped);
+			source_width = member_node->range_left - member_node->range_right + 1;
+
+			int dims = range->type == AST_MULTIRANGE ? GetSize(range->children) : 1;
+			stride = source_width;
```

Comment:
Can you add a comment explaining what's going on here with `stride`? My understanding is that this is an optimization because we know that `shift_expr` can only be multiples of `stride`, where `stride` is the number of entries in the slice of the LHS member. You can probably describe this better than I can.

Reply:
Yes, you are correct. I have made an attempt at explaining this now.

Interestingly, for the cases I've tried, ABC seems to be so smart that it can work out the same result without any CASE optimization, but I think it's nice to be able to generate a much smaller AST.

P.S. Before I noticed your review, I also fixed a mistake in the code above.